toast-tk-runtime/src/main/java/io/toast/tk/runtime/report/XmlPrettierHelper.java

```diff
@@ -0,0 +1,104 @@
+package io.toast.tk.runtime.report;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class XmlPrettierHelper {
+
+	static String getPrettySentenceText(String str) {
+		if(str.contains("*")) {
+			String separator = "\\*";
+			String[] lines = str.split(separator);
+			List<String> res = getPrettyXmlText(lines);
+			String resultat = String.join("*", res);
+			if(lines.length % 2  == 0) {
+				resultat += "*";
+			}
+			return resultat;
+		} else {
+			return getPrettyXmlText(str);
+		}
+	}
+	static List<String> getPrettyXmlText(String[] lines) {
+		List<String> res = new ArrayList<String>();
+		for(String line : lines) {
+			if((line.indexOf("<") < line.indexOf(">") + 1) 
+				&& line.contains("<") && line.contains(">") ) {
+				res.add(System.lineSeparator() + getPrettyXmlText(line)
+						+ System.lineSeparator());
+			} else {
+				if(!"".equals(line)) {
+					res.add(line);
+				}
+			}
+		}
+		return res;
+	}
+	static String getPrettyXmlText(String str) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 52 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)